userspace/otpilot/src/main.rs

```diff
@@ -236,51 +278,40 @@ impl<'a> SpiProcessor<'a> {
     }
 
     fn spi_write_to_buf<H>(&mut self, header: &H, data: &[u8], mut buf: &mut[u8])
-    -> Result<usize, FromWireError>
+    -> SpiProcessorResult<usize>
     where H: flash::SpiHeader + ToWire {
-        let mut console = Console::new();
         let mut tx_cursor = SpiutilsCursor::new(&mut buf);
-        if let Err(why) = header.to_wire(&mut tx_cursor) {
-            writeln!(console, "Host: Could not store header: {:?}", why);
-            return Err(FromWireError::OutOfRange);
-        }
+        header.to_wire(&mut tx_cursor)?;
         if header.get_opcode().has_dummy_byte() {
             // Skip dummy byte
-            if let Err(why) = tx_cursor.write_bytes(&[1; 0]) {
-                writeln!(console, "Host: Could not store dummy byte: {:?}", why);
-                return Err(FromWireError::OutOfRange);
-            }
+            tx_cursor.write_bytes(&[1; 0])
```

Comment:
Was this supposed to be a length-1 array containing a zero byte rather than an empty slice?

Reply:
No, we're writing a 1 byte array with content "0x0". That's the "dummy" byte that needs to be sent. Did I get the syntax wrong?

Reply:
Yes, a 1-byte array with content "0x0" is `&[0; 1]`.

Reply:
Oops ... jeez. What a bug. Thanks for noticing that. That would have caused some head-scratching at some point! Fixed in separate commit in same PR.